### DISCUSSION TITLE
How to find build_password_encrypted and build_key

### DISCUSSION THREAD

**Discussion by levensailor** [vmware/packer-examples-for-vsphere#746] [Help] [answered]
I changed the build_username and build_password, how do i get the build_password_encrypted and build_key to input into build.pkrvars.hcl? I'm doing an ubuntu 22.04 lts build with ansible provisioner and the provisioner fails. I've verified I can ssh to the box with build_username and build_password. 

here is what i'm seeing - it gets to the provisioner ansible step and fails. 

    vsphere-iso.linux-ubuntu:
    vsphere-iso.linux-ubuntu: TASK [Gathering Facts] *********************************************************
    vsphere-iso.linux-ubuntu: task path: /Users/<sensitive>/packer-examples-for-vsphere/builds/linux/ubuntu/22-04-lts/<sensitive>/main.yml:2
    vsphere-iso.linux-ubuntu: fatal: [default]: FAILED! => {
    vsphere-iso.linux-ubuntu:     "msg": "failed to transfer file to /Users/<sensitive>/.<sensitive>/tmp/<sensitive>-local-15047metwzbqm/tmp2jng6f0r /home/<sensitive>/.<sensitive>/tmp/<sensitive>-tmp-1697600373.529771-15050-33799668839196/AnsiballZ_setup.py:\n\nExecuting: program /usr/bin/ssh host 127.0.0.1, user (unspecified), command sftp\nOpenSSH_9.0p1, LibreSSL 3.3.6\r\ndebug1: Reading configuration data /Users/<sensitive>/.ssh/config\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 21: include /etc/ssh/ssh_config.d/* matched no files\r\ndebug1: /etc/ssh/ssh_config line 54: Applying options for *\r\ndebug2: resolve_canonicalize: hostname 127.0.0.1 is address\r\ndebug3: expanded UserKnownHostsFile '~/.ssh/known_hosts' -> '/Users/<sensitive>/.ssh/known_hosts'\r\ndebug3: expanded UserKnownHostsFile '~/.ssh/known_hosts2' -> '/Users/<sensitive>/.ssh/known_hosts2'\r\ndebug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 15054\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\nscp: debug2: Remote version: 3\r\nscp: debug2: Server supports extension \"posix-rename@openssh.com\" revision 1\r\nscp: debug2: Server supports extension \"statvfs@openssh.com\" revision 2\r\nscp: debug2: Server supports extension \"fstatvfs@openssh.com\" revision 2\r\nscp: debug2: Server supports extension \"hardlink@openssh.com\" revision 1\r\nscp: debug2: Server supports extension \"fsync@openssh.com\" revision 1\r\nscp: debug2: Server supports extension \"lsetstat@openssh.com\" revision 1\r\nscp: debug2: Server supports extension \"limits@openssh.com\" revision 1\r\nscp: debug2: Server supports extension \"expand-path@openssh.com\" revision 1\r\nscp: debug3: Sent message limits@openssh.com I:1\r\nscp: debug3: Received limits reply T:201 I:1\r\nscp: debug2: Sending SSH2_FXP_STAT \"/home/<sensitive>/.<sensitive>/tmp/<sensitive>-tmp-1697600373.529771-15050-33799668839196/AnsiballZ_setup.py\"\r\nscp: debug3: Sent message fd 6 T:17 I:2\r\nscp: debug1: stat remote: No such file or directory\r\nscp: debug3: source_sftp: copying local /Users/<sensitive>/.<sensitive>/tmp/<sensitive>-local-15047metwzbqm/tmp2jng6f0r to remote /home/<sensitive>/.<sensitive>/tmp/<sensitive>-tmp-1697600373.529771-15050-33799668839196/AnsiballZ_setup.py\r\nscp: debug2: do_upload: upload local \"/Users/<sensitive>/.<sensitive>/tmp/<sensitive>-local-15047metwzbqm/tmp2jng6f0r\" to remote \"/home/<sensitive>/.<sensitive>/tmp/<sensitive>-tmp-1697600373.529771-15050-33799668839196/AnsiballZ_setup.py\"\r\nscp: debug2: Sending SSH2_FXP_OPEN \"/home/<sensitive>/.<sensitive>/tmp/<sensitive>-tmp-1697600373.529771-15050-33799668839196/AnsiballZ_setup.py\"\r\nscp: debug3: Sent dest message SSH2_FXP_OPEN I:3 P:/home/<sensitive>/.<sensitive>/tmp/<sensitive>-tmp-1697600373.529771-15050-33799668839196/AnsiballZ_setup.py M:0x001a\r\nscp: debug3: Sent message SSH2_FXP_WRITE I:5 O:0 S:32768\r\nscp: debug3: SSH2_FXP_STATUS 0\r\nscp: debug3: In write loop, ack for 5 32768 bytes at 0\r\nscp: debug3: Sent message SSH2_FXP_WRITE I:6 O:32768 S:32768\r\nscp: debug3: Sent message SSH2_FXP_WRITE I:7 O:65536 S:32768\r\nscp: debug3: Sent message SSH2_FXP_WRITE I:8 O:98304 S:32768\r\nscp: debug3: Sent message SSH2_FXP_WRITE I:9 O:131072 S:32768\r\nscp: debug3: Sent message SSH2_FXP_WRITE I:10 O:163840 S:32768\r\nscp: debug3: Sent message SSH2_FXP_WRITE I:11 O:196608 S:32768\r\nscp: debug3: Sent message SSH2_FXP_WRITE I:12 O:229376 S:32768\r\nscp: debug3: Sent message SSH2_FXP_WRITE I:13 O:262144 S:32768\r\nscp: debug3: Sent message SSH2_FXP_WRITE I:14 O:294912 S:2934\r\nscp: debug3: SSH2_FXP_STATUS 0\r\nscp: debug3: In write loop, ack for 6 32768 bytes at 32768\r\nscp: debug3: SSH2_FXP_STATUS 0\r\nscp: debug3: In write loop, ack for 7 32768 bytes at 65536\r\nscp: debug3: SSH2_FXP_STATUS 0\r\nscp: debug3: In write loop, ack for 8 32768 bytes at 98304\r\nscp: debug3: SSH2_FXP_STATUS 0\r\nscp: debug3: In write loop, ack for 9 32768 bytes at 131072\r\nscp: debug3: SSH2_FXP_STATUS 0\r\nscp: debug3: In write loop, ack for 10 32768 bytes at 163840\r\nscp: debug3: SSH2_FXP_STATUS 0\r\nscp: debug3: In write loop, ack for 11 32768 bytes at 196608\r\nscp: debug3: SSH2_FXP_STATUS 0\r\nscp: debug3: In write loop, ack for 12 32768 bytes at 229376\r\nscp: debug3: SSH2_FXP_STATUS 0\r\nscp: debug3: In write loop, ack for 13 32768 bytes at 262144\r\nscp: debug3: SSH2_FXP_STATUS 0\r\nscp: debug3: In write loop, ack for 14 2934 bytes at 294912\r\nscp: debug3: Sent message SSH2_FXP_CLOSE I:4\r\nscp: debug3: SSH2_FXP_STATUS 0\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Control master terminated unexpectedly\r\n"
    vsphere-iso.linux-ubuntu: }
    vsphere-iso.linux-ubuntu:
    vsphere-iso.linux-ubuntu: PLAY RECAP *********************************************************************
    vsphere-iso.linux-ubuntu: default                    : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
    vsphere-iso.linux-ubuntu:
==> vsphere-iso.linux-ubuntu: Error executing Ansible: Non-zero exit status: exit status 2
==> vsphere-iso.linux-ubuntu: Step "StepProvision" failed

**Comment by tenthirtyam** [comment, marked as answer]
Please refer to the OpenSSH section:

https://vmware-samples.github.io/packer-examples-for-vsphere/getting-started/requirements/#platform

**Reply by tenthirtyam** [reply]
Ref: #747
